src/utils/string.test.ts

```diff
@@ -0,0 +1,28 @@
+import { it, describe, expect } from "vitest";
```

Comment:
I added this test file. Use it to test the refactor of parseNumericString so that it fulfils the behaviour I listed in the comment.